### PR TITLE
cucumber-expressions: {word}-expression with unicode fix

### PR DESCRIPTION
**Pull request by savkk**
## Summary

- fix {word}-expression using Unicode characters.
- fix line separator in ExpressionExamplesTest.

## Details
I think the summary is detailed enough :)

## Motivation and Context

I have a problem when using {word}-expression with russian words. 

## How Has This Been Tested?

I've written unit-tests. An I'm testing it in my bdd-project. 

## Types of changes

- [x ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/ExpressionExamplesTest.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public ExpressionExamplesTest(String expressionString, String text, String expec
         String[] chunks = s.split("---");
         for (String chunk : chunks) {
             chunk = chunk.trim();
-            data.add(chunk.split("\n"));
+            data.add(chunk.split(System.lineSeparator()));
```
Comment (mpkorstanje) [inline cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/ExpressionExamplesTest.java]: Please ensure the examples.txt file has unix line endings and this shouldn't be needed.

Comment (aslakhellesoy) [reply cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/ExpressionExamplesTest.java]: Looks good - I’ll merge this when the other implementations have a similar fix.

Comment (savkk) [reply cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/ExpressionExamplesTest.java]: @mpkorstanje, I use Windows as default OS. And I've faced a problem when tried to run tests that were in your repo. I think Files.readAllBytes(Path p) adds line separator depending on the OS that it's executed on.
So I can't use unix line endings if I use Windows. I guess, it won't be convenient for other users too.

Comment (savkk) [reply cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/ExpressionExamplesTest.java]: @aslakhellesoy, I don't really understand what exactly do you mean by "other implementations".

Comment (aslakhellesoy) [reply cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/ExpressionExamplesTest.java]: JavaScript, Go and Ruby

**Comment by aslakhellesoy** [comment]
Hi @savkk,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by lsuski** [comment]
This broke cucumber on Android because of `java.lang.IllegalArgumentException: Unsupported flags: 256`. I''l check if can find workaround for this

**Comment by lsuski** [comment]
Although this flag exists in Android and docs says that it has no effect on Android but can be passed it is omitted in source code and passing it throws exception.

**Pull request merged**